demoapp/testimony_summary.py

```diff
@@ -0,0 +1,204 @@
+
+'''
+Please pass in open ai api key and bill number as inputs. As of now this program can only work with 3 bills (H3069, H3121, H711)
+
+python testimony_summary.py <<open ai secret key>> <<bill number>>
+
+
+Example command to run this program
+
+python testimony_summary.py sk-my_key H711
+
+'''
+
+import textwrap
+import pandas as pd
+import tiktoken
+import os
+import json
+import sys
+
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.prompts import PromptTemplate
+from langchain.chains import LLMChain
+from langchain.docstore.document import Document
+from langchain.vectorstores import Chroma
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.chains.summarize import load_summarize_chain
+from langchain.schema.runnable import RunnablePassthrough
+from langchain.schema.output_parser import StrOutputParser
+
+from langchain_community.chat_models import ChatOpenAI
+from ipdb import set_trace as st
+from tagging import *
+
+try: 
+    with open('generated_summary.json', 'r') as f: SUMMARY_CACHE = json.load(f)
+except Exception as e: SUMMARY_CACHE = {}
+
+open_ai_key, bill_id = sys.argv[1:]
+os.environ['OPENAI_API_KEY'] = open_ai_key 
+MODEL_NAME = 'gpt-4-1106-preview'
+MAX_TOKENS = 4097
+
+mgl_df = pd.read_csv("./12_bills_with_mgl_more_sections.csv")
+summaries = pd.read_csv('./generated_bills.csv')
+testimonies = pd.read_csv('./maple_testimonies.csv')
+
+text_splitter = CharacterTextSplitter.from_tiktoken_encoder(model_name = MODEL_NAME)
+
+bills_to_select = {
+    '#H3121': 'An Act relative to the open meeting law',
+    '#S2064': 'An Act extending the public records law to the Governor and the Legislature',
+    '#H711': 'An Act providing a local option for ranked choice voting in municipal elections',
+    '#S1979': 'An Act establishing a jail and prison construction moratorium',
+    '#H489': 'An Act providing affordable and accessible high-quality early education and care to promote child development and well-being and support the economy in the Commonwealth',
+    '#S2014': 'An Act relative to collective bargaining rights for legislative employees',
+    '#S301': 'An Act providing affordable and accessible high quality early education and care to promote child development and well-being and support the economy in the Commonwealth',
+    '#H3069': 'An Act relative to collective bargaining rights for legislative employees',
+    '#S433': 'An Act providing a local option for ranked choice voting in municipal elections',
+    '#H400': 'An Act relative to vehicle recalls',
+    '#H538': 'An Act to Improve access, opportunity, and capacity in Massachusetts vocational-technical education',
+    '#S257': 'An Act to end discriminatory outcomes in vocational school admissions'
+}
+
+def find_bills(bill_number, df = mgl_df):
```

Comment:
Please add typehints to all function sigs

Reply:
Sure thing